yext/react-site-search-starter
# Change usePageSetupEffect to useInitialSearch

Rename `usePageSetupEffect` to `useInitialSearch`. Rework it have Headless clear the previous searcher's state and add a config for the initial search that accepts a `verticalKey`, `defaultInitialSearch`, `verticalLimit`, and/or `universalLimit`. The default initial search should only be used if there is no user-entered query (including an empty string query) from the previous searcher (i.e. there is no query param stored in the URL). If used, the default initial search query should not be stored in the URL. So, the `Navigation` component links were updated to only add the query param if there is a previous query.

Also, make a tweak to the styling for the collapse filters button on mobile to match the mocks.

J=SLAP-1910
TEST=manual

- Check that specifying universal or vertical limits displays the correct number of results
- Check that the `defaultInitialSearch` is only used if there is no previous query persisted in the URL
- Check that the `defaultInitialSearch` is not persisted in the URL
- Check that a user-entered query is used on a new searcher even if that searcher has a `defaultInitialSearch` defined
- Check that if there is no `defaultInitialSearch` or query param in the URL, an empty string query is used
